src/userInputParameters/load_BC_list.cc

```diff
@@ -1,102 +1,78 @@
+// ------------------------------------------------------------------------
+// Method to extract the list of boundary conditions from input parameters
+// and store them in BC_list object
+// ------------------------------------------------------------------------
+
 #include "../../include/userInputParameters.h"
 
-// ==========================================================================================
-// Method to extract the list of boundary conditions
-// ==========================================================================================
 template <int dim>
-void
-userInputParameters<dim>::load_BC_list(std::vector<std::string> list_of_BCs)
+void userInputParameters<dim>::load_BC_list(std::vector<std::string> list_of_BCs)
 {
-  // Load the BC information from the strings into a varBCs object
-  // Move this to a new method and write a unit test for it!!!!
-
-  std::vector<std::string> temp;
-
-  for (unsigned int i = 0; i < list_of_BCs.size(); i++)
-    {
-      varBCs<dim> newBC;
-      temp = dealii::Utilities::split_string_list(list_of_BCs[i]);
+    // Loop over the list of boundary conditions specified in parameters
+    // and provided in the input list_of_BCs. Process the BCs and place
+    // them into the vector BC_list
+    std::vector<std::string> temp;
+    for (unsigned int i=0; i<list_of_BCs.size(); i++){
+        
+        // Ensure all variables have BCs specified in parameters.prm
+        if (list_of_BCs[i] == ""){
+            std::cout << "Error: Boundary condition not specified." << std::endl;
+            abort();
```

Comment:
I think it might be better to use `AssertThrow(list_of_BCs[i].empty(), ExcMessage("Error Message")`

The only thing is that this assert will only show up in debug mode.

Reply:
I revisited the deal.II documentation and it says AssertThrow() actually does work for both debug and release mode (I'm not sure why it wasn't working for me before). I made the changes and made a new commit.